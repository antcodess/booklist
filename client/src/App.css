* {
  box-sizing: border-box;
}

@font-face {
  font-family: "Hurme Geometric Sans 3 W03 Bold";
  src: url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.eot"); /* IE9*/
  src: url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.eot?#iefix") format("embedded-opentype"), /* IE6-IE8 */
  url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.woff2") format("woff2"), /* chrome、firefox */
  url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.woff") format("woff"), /* chrome、firefox */
  url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.ttf") format("truetype"), /* chrome、firefox、opera、Safari, Android, iOS 4.2+*/
  url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.svg#Hurme Geometric Sans 3 W03 Bold") format("svg"); /* iOS 4.1- */
}

@font-face {
  font-family: "Cera GR Bold";
  src: url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.eot"); /* IE9*/
  src: url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.eot?#iefix") format("embedded-opentype"), /* IE6-IE8 */
  url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.woff2") format("woff2"), /* chrome、firefox */
  url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.woff") format("woff"), /* chrome、firefox */
  url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.ttf") format("truetype"), /* chrome、firefox、opera、Safari, Android, iOS 4.2+*/
  url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.svg#Cera GR Bold") format("svg"); /* iOS 4.1- */
}

@font-face {
  font-family: "Cera GR Light";
  src: url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.eot"); /* IE9*/
  src: url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.eot?#iefix") format("embedded-opentype"), /* IE6-IE8 */
  url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.woff2") format("woff2"), /* chrome、firefox */
  url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.woff") format("woff"), /* chrome、firefox */
  url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.ttf") format("truetype"), /* chrome、firefox、opera、Safari, Android, iOS 4.2+*/
  url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.svg#Cera GR Light") format("svg"); /* iOS 4.1- */
}

@font-face {
  font-family: "Hurme Geometric Sans 3 W03 Lt";
  src: url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.eot"); /* IE9*/
  src: url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.eot?#iefix") format("embedded-opentype"), /* IE6-IE8 */
  url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.woff2") format("woff2"), /* chrome、firefox */
  url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.woff") format("woff"), /* chrome、firefox */
  url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.ttf") format("truetype"), /* chrome、firefox、opera、Safari, Android, iOS 4.2+*/
  url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.svg#Hurme Geometric Sans 3 W03 Lt") format("svg"); /* iOS 4.1- */
}

h1 { 
  font-family: "Hurme Geometric Sans 3 W03 Bold";
  font-size: 3.125rem;
  line-height: 0.65;
}

h2 {
  font-family: "Hurme Geometric Sans 3 W03 Bold";
  font-size: 2.5rem;
  line-height: 1;
}

h3 {
  font-family: "Hurme Geometric Sans 3 W03 Bold";
  font-size: 1.56rem;
}

h4 {
  font-family: "Cera GR Bold";
  font-size: .98rem;
}

button {
  width: auto;
  display: flex;
  justify-content: center;
  height: 40px;
  padding: 0 30px;
  background-color: black;
  color: white;
  font-family: "Hurme Geometric Sans 3 W03 Bold";
  font-size: 0.84rem;
  text-transform: uppercase;
  border-radius: 5px;
  border: none;
  cursor: pointer;
}

button:hover {
  background-color: #0080DE;
}

span {
  font-family: "Hurme Geometric Sans 3 W03 Lt";
}

ul li {
  font-family: "Cera GR Light";
  font-size: 0.94rem;
  color: black;
}

img {
  border-radius: 10px;
}

.main-content {
  margin: 0 auto;
  text-align: center;
  padding: 0px 100px;
  max-width: 1720px;
  min-height: 70vh;
}

@media only screen and (min-width: 950px) {
  .regular-sign-in {
    width: 40%;
  }
}

@media only screen and (min-width: 1000px) {
  .slider .image-placeholder .custom-button {
    font-size: 1.0rem;
  }
}

@media only screen and (min-width: 1100px) {
  .slider .image-placeholder .custom-button {
    font-size: 1.1rem;
  }
}

@media only screen and (min-width: 1200px) {
  .slider .image-placeholder .custom-button {
    font-size: 1.2rem;
  }
}

@media only screen and (min-width: 1500px) {
  .slider .image-placeholder .custom-button {
    font-size: 1.6rem;
  }
}

@media only screen and (max-width: 1500px) {
  .directory-items-container .directory-items .slick-prev, .directory-items-container .directory-items .slick-next {
    top: 125px;
  }
}

@media only screen and (max-width: 1100px) {
  .directory-items-container .directory-items .slick-prev, .directory-items-container .directory-items .slick-next {
    top: 95px;
  }
}

@media only screen and (min-width: 1100px) {
  .regular-sign-in, .google-sign-in {
    width: 50%;
  }
}

@media only screen and (max-width: 940px) {
  .category-items-container .category-items {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

@media only screen and (max-width: 882px) {
  .buttons {
    flex-direction: column;
  }
  .main-content {
    padding: 0px 50px;
  }
}


@media only screen and (max-width: 830px) {
  .footer-top {
    flex-direction: column;
  }

  .footer-bottom {
    justify-content: center;
  }

  .footer-top h1 {
    margin-bottom: 30px;
  }

  .slider .image-placeholder .custom-button {
    font-size: 0.84rem;
  }

}

@media only screen and (max-width: 820px) {
  .product-item .product-item-top {
    justify-content: center;
    flex-direction: column;
  }

  .product-item .product-item-top .product-image {
    width: 100%;
  }

  .product-item .product-item-top .product-details {
    margin-left: 0px;
    width: 100%;
  }

  .product-item .product-item-top .product-details h2 {
    margin-top: 20px;
    justify-content: center;
    text-align: center;
  }

  .product-item .product-item-top .product-details h3{
    text-align: center;
  }

  .product-item .product-item-top .product-details .product-details-text {
    justify-content: center;
  }

  .product-item .product-item-top .product-details .product-price {
    justify-content: center;
  }

}

@media only screen and (max-width: 720px) {
  .navigation .options .option {
    display: none;
  }

  .nav-icon {
    display: flex;
  }

  /* .slider .image-placeholder .custom-button {
    font-size: 0.64rem;
  } */
  .sign-in-and-sign-up {
    flex-direction: column;
    justify-content: center;
  }
  .sign-in {
    width: 100%;
    margin-bottom: 40px;
  }

  .sign-up {
    width: 100%;
  }

  .sign-up .custom-button {
    width: 100%;
  }

  .main-content {
    width: auto;
    overflow:hidden;
    max-width: auto;
  }
}

@media only screen and (max-width: 700px) {
  .category-items-container .category-items {
    grid-template-columns: 1fr 1fr;
  }
}

@media only screen and (max-width: 670px) {
  .footer .footer-content .footer-top .footer-list li {
    border-left: none;
  }
}

@media only screen and (max-width: 650px) {
  .main-content {
    padding: 0px 20px;
  }
}

@media only screen and (max-width: 600px) {
  .directory-items-container .directory-items .slick-prev, .directory-items-container .directory-items .slick-next {
    top: 155px;
  }
  .directory-items-container h3 {
    justify-content: center;
  }

  .category-items-container .category-items {
    grid-template-columns: 1fr;
  }

  .directory-items-container .directory-items-title {
    justify-content: center;
  }

  .category-items-container h2 {
    text-align: center;
  }

  .slider .image-placeholder .custom-button {
    font-size: .64rem;
    padding: 0px 10px;
  }
}

@media only screen and (max-width: 500px) {
  .directory-items-container .directory-items .slick-prev, .directory-items-container .directory-items .slick-next {
    top: 130px;
  }

  .cart-icon {
    /* display: none; */
    margin-right: 2px;
  }

  .nav-dropdown {
    /* left: 198px; */
    left: 175px;
  }

  .nav-icon {
    /* top: 2px; */
    left: 206px;
    /* left: 230px; */
  }

  .category-items-container h2 {
    text-align: center;
  }

  .footer-top .footer-list  {
    flex-direction: column;
  }

  .footer .footer-content .footer-top .footer-list li {
    margin-bottom: 10px;
    padding-left: 0;
  }
}

@media only screen and (max-width: 480px) {
  .header h1 {
    margin-left: 0px;
  }

  .nav-dropdown {
    left: 176px;
  } 

  .nav-icon {
    left: 207px;
  }

  .slider .image-placeholder .custom-button {
    /* font-size: 3px;
    padding: 0px 36px; */
    display: none;
  }

  .product-item .product-item-bottom .product-description .product-description-text p {
    text-align: center;
  }

  .main-content {
    padding: 0px 5px;
  }

  .footer .footer-content .footer-bottom .footer-bottom-text {
    text-align: center;
  }
}

@media only screen and (max-width: 400px) {
  .directory-items-container .directory-items .slick-prev, .directory-items-container .directory-items .slick-next {
    top: 83px;
  }
  .header h1 {
    font-size: 2.5rem;
  }
  .nav-icon {
    background-size: 30px;
    top: 11px;
    left: 165px;
  }
  .nav-dropdown {
    top: 42px;
    left: 130px;
    z-index: 5;
  }
}
