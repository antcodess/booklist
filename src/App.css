* {
  box-sizing: border-box;
}

@font-face {
  font-family: "Hurme Geometric Sans 3 W03 Bold";
  src: url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.eot"); /* IE9*/
  src: url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.eot?#iefix") format("embedded-opentype"), /* IE6-IE8 */
  url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.woff2") format("woff2"), /* chrome、firefox */
  url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.woff") format("woff"), /* chrome、firefox */
  url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.ttf") format("truetype"), /* chrome、firefox、opera、Safari, Android, iOS 4.2+*/
  url("/fonts/hurmebold/1b691f4637dd58c971711cd611f5f6bd.svg#Hurme Geometric Sans 3 W03 Bold") format("svg"); /* iOS 4.1- */
}

@font-face {
  font-family: "Cera GR Bold";
  src: url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.eot"); /* IE9*/
  src: url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.eot?#iefix") format("embedded-opentype"), /* IE6-IE8 */
  url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.woff2") format("woff2"), /* chrome、firefox */
  url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.woff") format("woff"), /* chrome、firefox */
  url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.ttf") format("truetype"), /* chrome、firefox、opera、Safari, Android, iOS 4.2+*/
  url("/fonts/cerabold/d191325cfe4f374d6f43147fd57ab2d4.svg#Cera GR Bold") format("svg"); /* iOS 4.1- */
}

@font-face {
  font-family: "Cera GR Light";
  src: url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.eot"); /* IE9*/
  src: url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.eot?#iefix") format("embedded-opentype"), /* IE6-IE8 */
  url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.woff2") format("woff2"), /* chrome、firefox */
  url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.woff") format("woff"), /* chrome、firefox */
  url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.ttf") format("truetype"), /* chrome、firefox、opera、Safari, Android, iOS 4.2+*/
  url("/fonts/ceralight/4c23d1c747b7c11cecbcc0bc4ecbed1a.svg#Cera GR Light") format("svg"); /* iOS 4.1- */
}

@font-face {
  font-family: "Hurme Geometric Sans 3 W03 Lt";
  src: url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.eot"); /* IE9*/
  src: url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.eot?#iefix") format("embedded-opentype"), /* IE6-IE8 */
  url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.woff2") format("woff2"), /* chrome、firefox */
  url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.woff") format("woff"), /* chrome、firefox */
  url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.ttf") format("truetype"), /* chrome、firefox、opera、Safari, Android, iOS 4.2+*/
  url("/fonts/hurmelight/471a8cf5be29aa662f525c3b132187d7.svg#Hurme Geometric Sans 3 W03 Lt") format("svg"); /* iOS 4.1- */
}

h1 { 
  font-family: "Hurme Geometric Sans 3 W03 Bold";
  font-size: 3.125rem;
  line-height: 0.65;
}

h3 {
  font-family: "Hurme Geometric Sans 3 W03 Bold";
  font-size: 1.56rem;
}

h4 {
  font-family: "Cera GR Bold";
  font-size: .98rem;
}

button {
  width: auto;
  display: flex;
  justify-content: center;
  height: 40px;
  padding: 0 30px;
  background-color: black;
  color: white;
  font-family: "Hurme Geometric Sans 3 W03 Bold";
  font-size: 0.84rem;
  text-transform: uppercase;
  border-radius: 5px;
  border: none;
  cursor: pointer;
}

button:hover {
  background-color: #0080DE;
}

span {
  font-family: "Hurme Geometric Sans 3 W03 Lt";
}

.main-content {
  margin: 0 auto;
  text-align: center;
  padding: 0px 100px;
}

@media only screen and (max-width: 1500px) {
  .directory-items-container .directory-items .slick-prev, .directory-items-container .directory-items .slick-next {
    top: 125px;
  }
}

@media only screen and (max-width: 1100px) {
  .directory-items-container .directory-items .slick-prev, .directory-items-container .directory-items .slick-next {
    top: 95px;
  }
}

@media only screen and (max-width: 600px) {
  .directory-items-container .directory-items .slick-prev, .directory-items-container .directory-items .slick-next {
    top: 155px;
  }
}

@media only screen and (max-width: 500px) {
  .directory-items-container .directory-items .slick-prev, .directory-items-container .directory-items .slick-next {
    top: 130px;
  }
}

@media only screen and (max-width: 400px) {
  .directory-items-container .directory-items .slick-prev, .directory-items-container .directory-items .slick-next {
    top: 83px;
  }
}
