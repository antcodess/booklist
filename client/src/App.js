import React, { useEffect, lazy, Suspense } from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import Header from './components/Header/Header';
import Footer from './components/Footer/Footer';
import Alert from './components/Alert/Alert';
import Spinner from './components/Spinner/Spinner';

import { fetchProductsStart, fetchCategoriesStart } from './redux/shop/shop.actions';
import { selectCurrentUser } from './redux/user/user.selectors';
import { selectCartHidden } from './redux/cart/cart.selectors';
import { selectNavHidden } from './redux/nav/nav.selectors';

import { checkUserSession } from './redux/user/user.actions';
import { toggleCartHidden } from './redux/cart/cart.actions';
import { toggleNavHidden } from './redux/nav/nav.actions';

import './App.css';

const HomePageContainer = lazy(() => import('./pages/HomePage/HomePageContainer'));
const ProductsPageContainer = lazy(() => import('./pages/ProductsPage/ProductsPageContainer'));
const ShopPageContainer = lazy(() => import('./pages/ShopPage/ShopPageContainer'));
const CategoriesPageContainer = lazy(() => import('./pages/CategoriesPage/CategoriesPageContainer'));
const CheckoutPageContainer = lazy(() => import('./pages/CheckoutPage/CheckoutPageContainer'));
const AuthorPageContainer = lazy(() => import('./pages/AuthorsPage/AuthorsPageContainer'));

const SignInAndSignUpPage = lazy(() => import('./pages/SignInAndSignUpPage/SignInAndSignUpPage'));
const AccountPage = lazy(() => import('./pages/AccountPage/AccountPage'));
const ThankYouPage = lazy(() => import('./pages/ThankYouPage/ThankYouPage'));

const App = ({ 
  fetchProductsStart, 
  fetchCategoriesStart, 
  currentUser, 
  checkUserSession, 
  hidden, 
  toggleCartHidden,
  navHidden,
  toggleNavHidden }) => {

  useEffect(() => {
    fetchProductsStart();
    fetchCategoriesStart();
    checkUserSession();

  }, [fetchProductsStart, fetchCategoriesStart, checkUserSession])

  const onAppClick = (event) => {
    const className = event.target.className;
    if (!hidden) {
      if (className !== 'arrow' && 
          className !== 'value' &&
          className !== 'shopping-icon') {
        toggleCartHidden();
      } 
    }
    if (!navHidden) {
      if (className !== 'nav-option' && className !== 'nav-icon') {
        toggleNavHidden();
      }
    }
  };

  return (
    <React.Fragment>
      <Alert />
      <div className="main-content" onClick={(event) => onAppClick(event)}>
        <Header />
          <Switch>
            <Suspense fallback={<Spinner />}>
              <Route exact path='/' component={HomePageContainer} />
              <Route path='/products' component={ProductsPageContainer} />
              <Route exact path='/shop' component={ShopPageContainer} />
              <Route exact path='/categories' component={ShopPageContainer} />
              <Route path='/categories/:categoryTitle' component={CategoriesPageContainer} />
              <Route exact path='/checkout' component={CheckoutPageContainer} />
              <Route path='/authors/:authorName' component={AuthorPageContainer} />
              <Route 
                exact 
                path='/signin' 
                render={() => currentUser ? (<Redirect to='/' />) : (<SignInAndSignUpPage />)} 
              />
              <Route 
                exact 
                path='/account' 
                render={() => !currentUser ? (<Redirect to='/signin' />) : (<AccountPage />)} 
              />
              <Route exact path='/thank-you' component={ThankYouPage} />
            </Suspense>
          </Switch>
      </div>
      <Footer />
    </React.Fragment>
  );
};

const mapStateToProps = createStructuredSelector({
  currentUser: selectCurrentUser,
  hidden: selectCartHidden,
  navHidden: selectNavHidden
});

const mapDispatchToProps = dispatch => ({
  fetchProductsStart: () => dispatch(fetchProductsStart()),
  fetchCategoriesStart: () => dispatch(fetchCategoriesStart()),
  checkUserSession: () => dispatch(checkUserSession()),
  toggleCartHidden: () => dispatch(toggleCartHidden()),
  toggleNavHidden: () => dispatch(toggleNavHidden())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
