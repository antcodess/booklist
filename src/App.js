import React from 'react';
import { Route, Switch } from 'react-router-dom';

import HomePage from './pages/HomePage/Homepage';
import ProductsPage from './pages/ProductsPage/ProductsPage';
import Header from './components/Header/Header';
import Footer from './components/Footer/Footer';

import './App.css';

const App = () => {
  return (
    <React.Fragment>
      <div className="main-content">
        <Header />
        <Switch>
          <Route exact path='/' component={HomePage} />
          <Route path='/products' component={ProductsPage} />
        </Switch>
      </div>
      <Footer />
    </React.Fragment>
  );
};


export default App;
