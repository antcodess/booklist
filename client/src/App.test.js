import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';
import { mapDispatchToProps } from './App';

describe('App Component', () => {
  let wrapper;
  let wrapper2;
  let mockCurrentUser; 
  let mockHidden;
  let mockNavHidden; 
  let mockFetchProductsStart; 
  let mockFetchCategoriesStart;
  let mockCheckUserSession;
  let mockToggleCartHidden;
  let mockToggleNavHidden;
  let mockOnAppClick;
  let mockEvent;
  let dummyFunction = jest.fn();
  
  beforeEach(() => {

    mockEvent = {
      target: {
        className: 'test-class'
      }
    };
    
    mockCurrentUser = { id: 1, name: 'Sample User'};
    mockHidden = true;
    mockNavHidden = true;
    mockFetchProductsStart = jest.fn();
    mockFetchCategoriesStart = jest.fn();
    mockCheckUserSession = jest.fn();
    mockToggleCartHidden = jest.fn();
    mockToggleNavHidden = jest.fn();
    mockOnAppClick = jest.fn(mockEvent);

    const mockProps = {
      currentUser: mockCurrentUser, 
      hidden: mockHidden, 
      navHidden: mockNavHidden, 
      fetchProductsStart: mockFetchProductsStart, 
      fetchCategoriesStart: mockFetchCategoriesStart, 
      checkUserSession: mockCheckUserSession,
      toggleCartHidden: mockToggleCartHidden,
      toggleNavHidden: mockToggleNavHidden
    };

    const mockProps2 = {
      currentUser: null, 
      hidden: false, 
      navHidden: false, 
      fetchProductsStart: mockFetchProductsStart, 
      fetchCategoriesStart: mockFetchCategoriesStart, 
      checkUserSession: mockCheckUserSession,
      toggleCartHidden: mockToggleCartHidden,
      toggleNavHidden: mockToggleNavHidden
    };

    mockEvent = {
      target: {
        className: 'test-class'
      }
    };
    
    // Signed In User
    wrapper = shallow(<App { ...mockProps } />);
    // Signed Out User
    wrapper2 = shallow(<App { ...mockProps2 } />);
  });

  it('should render App Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch fetchProductsStart action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchProductsStart();
    expect(dispatch.mock.calls[0][0]).toEqual({type: 'FETCH_PRODUCTS_START'});
  });

  it('should dispatch fetchCategoriesStart action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchCategoriesStart();
    expect(dispatch.mock.calls[0][0]).toEqual({type: 'FETCH_CATEGORIES_START'});
  });

  it('should dispatch fetchCategoriesStart action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchCategoriesStart();
    expect(dispatch.mock.calls[0][0]).toEqual({type: 'FETCH_CATEGORIES_START'});
  });

  it('should dispatch checkUserSession action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).checkUserSession();
    expect(dispatch.mock.calls[0][0]).toEqual({type: 'CHECK_USER_SESSION'});
  });

  it('should dispatch toggleCartHidden action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).toggleCartHidden();
    expect(dispatch.mock.calls[0][0]).toEqual({type: 'TOGGLE_CART_HIDDEN'});
  });

  it('should dispatch toggleNavHidden action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).toggleNavHidden();
    expect(dispatch.mock.calls[0][0]).toEqual({type: 'TOGGLE_NAV_HIDDEN'});
  });
});
