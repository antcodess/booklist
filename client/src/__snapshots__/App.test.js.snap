// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render App Component 1`] = `ShallowWrapper {}`;
